src/Microsoft.Maintainability.Analyzers/UnitTests/AvoidUnusedPrivateFieldsTests.Fixer.cs

```diff
@@ -12,22 +14,92 @@ public class AvoidUnusedPrivateFieldsFixerTests : CodeFixTestBase
     {
         protected override DiagnosticAnalyzer GetBasicDiagnosticAnalyzer()
         {
-            return new BasicAvoidUnusedPrivateFieldsAnalyzer();
+            return new AvoidUnusedPrivateFieldsAnalyzer();
         }
 
         protected override DiagnosticAnalyzer GetCSharpDiagnosticAnalyzer()
         {
-            return new CSharpAvoidUnusedPrivateFieldsAnalyzer();
+            return new AvoidUnusedPrivateFieldsAnalyzer();
         }
 
         protected override CodeFixProvider GetBasicCodeFixProvider()
         {
-            return new BasicAvoidUnusedPrivateFieldsFixer();
+            return new AvoidUnusedPrivateFieldsFixer();
         }
 
         protected override CodeFixProvider GetCSharpCodeFixProvider()
         {
-            return new CSharpAvoidUnusedPrivateFieldsFixer();
+            return new AvoidUnusedPrivateFieldsFixer();
+        }
+
+        [Fact]
+        public void CA1823CSharp()
+        {
+            VerifyCSharpFix(
+                @"  
+class C  
+{  
```

Comment:
What you are suggesting is a FixAll operation - given that John's fixer returns non-null batch fixer in GetFixAllProvider, it should be tested as the lightbulb will show Fix All options. However, I am not sure if there is an easy way to write FixAll tests in roslyn-analyzers repo (see [here](http://source.roslyn.io/#Roslyn.Services.Editor.CSharp.UnitTests/Diagnostics/RemoveUnnecessaryCast/RemoveUnnecessaryCastTests_FixAllTests.cs,16) for example tests in roslyn repo).